includes/class-debug.php

```diff
@@ -31,23 +31,20 @@ private static function get_file_path() {
 	/**
 	 * Initializes the WordPress Filesystem.
 	 *
-	 * @return WP_Filesystem_Base|false The filesystem object or false on failure.
+	 * @return WP_Filesystem_Direct|false The filesystem object or false on failure.
 	 */
 	private static function init_filesystem() {
-		global $wp_filesystem;
-
-		if ( ! $wp_filesystem ) {
-			require_once ABSPATH . 'wp-admin/includes/file.php';
-			WP_Filesystem();
-		}
-
-		return $wp_filesystem;
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		require_once ABSPATH . 'wp-admin/includes/class-wp-filesystem-base.php';
+		require_once ABSPATH . 'wp-admin/includes/class-wp-filesystem-direct.php';
+		WP_Filesystem();
+		return new Filesystem_Direct( false );
```

Comment:
Also, we can store the filesystem object to a static property to avoid re-running `WP_Filesystem()` and creating a new object each time.

```php
/**
 * The filesystem.
 *
 * @var Filesystem_Direct
 */
private static $filesystem;

// init_filesystem()
if ( ! self::$filesystem instanceof Filesystem_Direct ) {
	require_once ABSPATH . 'wp-admin/includes/file.php';
	require_once ABSPATH . 'wp-admin/includes/class-wp-filesystem-base.php';
	require_once ABSPATH . 'wp-admin/includes/class-wp-filesystem-direct.php';
	WP_Filesystem();
	self::$filesystem = new Filesystem_Direct( false );
}
return self::$filesystem;
```